From a9ebe831e1fb73019632a5285f84d07ec39ff6c2 Mon Sep 17 00:00:00 2001
Message-ID: <a9ebe831e1fb73019632a5285f84d07ec39ff6c2.1710883390.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1710883390.git.sharlatanus@gmail.com>
References: <cover.1710883390.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Tue, 19 Mar 2024 20:57:50 +0000
Subject: [PATCH 02/10] gnu: Add cl-trivial-utilities.

* gnu/packages/lisp-xyz.scm (cl-trivial-utilities,
ecl-trivial-utilities, sbcl-trivial-utilities): New variables.

Change-Id: I1fde88484add5b77cf93ccacafb17ac548929481
---
 gnu/packages/lisp-xyz.scm | 32 ++++++++++++++++++++++++++++++++
 1 file changed, 32 insertions(+)

diff --git a/gnu/packages/lisp-xyz.scm b/gnu/packages/lisp-xyz.scm
index cd8dd6bf14..c3d199a890 100644
--- a/gnu/packages/lisp-xyz.scm
+++ b/gnu/packages/lisp-xyz.scm
@@ -8092,6 +8092,38 @@ (define-public cl-trivial-utf-8
 (define-public ecl-trivial-utf-8
   (sbcl-package->ecl-package sbcl-trivial-utf-8))
 
+(define-public sbcl-trivial-utilities
+  (let ((commit "279ff255562628196942632c543d91c357067221")
+        (revision "0"))
+    (package
+      (name "sbcl-trivial-utilities")
+      (version (git-version "0.4.0" revision commit))
+      (source
+       (origin
+         (method git-fetch)
+         (uri (git-reference
+               (url "https://gitlab.com/ediethelm/trivial-utilities")
+               (commit commit)))
+         (file-name (git-file-name "trivial-utilities" version))
+         (sha256
+          (base32 "0k1xmn5f5dik7scadw0vyy67mik4ypnfqbhlv2vsg9afxzbpx2dz"))))
+      (build-system asdf-build-system/sbcl)
+      (inputs
+       (list sbcl-alexandria sbcl-iterate sbcl-closer-mop))
+      (home-page "https://gitlab.com/ediethelm/trivial-utilities")
+      (synopsis "Collection of useful types, functions and macros for Common Lisp")
+      (description
+       "This package provides a collection of types, functions and macros.  Some
+ of the functionality is implemented from Grham's On Lisp and Seibel's Practical
+ Common Lisp.")
+      (license license:expat))))
+
+(define-public cl-trivial-utilities
+  (sbcl-package->cl-source-package sbcl-trivial-utilities))
+
+(define-public ecl-trivial-utilities
+  (sbcl-package->ecl-package sbcl-trivial-utilities))
+
 (define-public sbcl-idna
   (package
     (name "sbcl-idna")
-- 
2.41.0

